pallets/interest-accrual/src/lib.rs

```diff
@@ -437,118 +436,137 @@ pub mod pallet {
 			T::Time::now().as_secs()
 		}
 
-		pub fn reference_interest_rate(interest_rate_per_year: T::InterestRate) -> DispatchResult {
-			let interest_rate_per_sec = unchecked_conversion(interest_rate_per_year)?;
-			Rates::<T>::try_mutate(|rates| {
-				let rate = rates
-					.iter_mut()
-					.find(|rate| rate.interest_rate_per_sec == interest_rate_per_sec);
-
-				match rate {
-					Some(rate) => Ok(rate.reference_count.ensure_add_assign(1)?),
-					None => {
-						Self::validate_interest_rate(interest_rate_per_year)?;
-
-						let new_rate = RateDetailsOf::<T> {
-							interest_rate_per_sec,
-							accumulated_rate: One::one(),
-							reference_count: 1,
-						};
-
-						rates
-							.try_push(new_rate)
-							.map_err(|_| Error::<T>::TooManyRates)?;
-
-						Ok(())
-					}
+		pub fn reference_interest_rate(
```

Comment:
This method was already pretty deeply nested and it's worse now. Might be worth trying to break out the guts of it into its own function that just mutates the vector or something?

Reply:
I guess it makes sense, but let's note that as a future clean up? Doesn't seem relevant particularly to this PR and this one is somewhat urgent.